services/property-svc/internal/property/queries/v1/get_properties_by_subject_type.go

```diff
@@ -84,6 +91,19 @@ func NewGetPropertiesQueryHandler() GetPropertiesQueryHandler {
 			}
 		}
 
+		checks := hwutil.Map(properties, func(p *models.PropertyWithConsistency) hwauthz.PermissionCheck {
+			return hwauthz.NewPermissionCheck(user, perm.PropertyCanUserGet, perm.Property(p.ID))
+		})
+
+		canGet, err := authz.BulkCheck(ctx, checks)
+		if err != nil {
+			return nil, err
+		}
+
+		properties = hwutil.Filter(properties, func(i int, _ *models.PropertyWithConsistency) bool {
+			return canGet[i]
+		})
+
```

Comment:
We should prefer [looking up all resources beforehand](https://buf.build/authzed/api/docs/main:authzed.api.v1#authzed.api.v1.PermissionsService.LookupResources) instead of filtering them afterward. This is also not an consistency issue because when SpiceDB is out of sync, the permission checks will fail.

Reply:
You are likely to have access to a large amount of properties. The approach you suggest does not make sense in that case.

And equally: We expect to actually filter properties out very rarely (currently, not at all, you can see all properties in your organization).

As the requirements and thus permissions change, it might make more sense to do your approach.

Here is a guide: https://authzed.com/docs/spicedb/modeling/protecting-a-list-endpoint